shared/lib-hoppy/hoppy/async_hoppy_client.py

```diff
@@ -0,0 +1,204 @@
+            f"correlation_id={correlation_id}")
+        self.responses[correlation_id] = None
+
+        try:
+            self.async_publisher.publish_message(body,
+                                                 BasicProperties(app_id=self.app_id,
+                                                                 content_type="application/json",
+                                                                 reply_to=self.reply_queue_properties.name,
+                                                                 correlation_id=correlation_id))
+        except Exception as e:
+            logging.warning(f"event=requestError "
+                            f"client={self.name} "
+                            f"id={request_id} "
+                            f"queue={self.request_queue_properties.name} "
+                            f"correlation_id={correlation_id} "
+                            f"error='{UNEXPECTED_PUBLISH_ERROR}: {e}'")
+            raise ResponseException(message=UNEXPECTED_PUBLISH_ERROR)
+
+        wait_for_response_time = None
+        while True:
+            response = self.responses.get(correlation_id)
+            if response is not None:
+                self._terminate_correlation_id(correlation_id)
+                if isinstance(response, json.JSONDecodeError):
+                    logging.warning(f"event=requestError "
+                                    f"client={self.name} "
+                                    f"id={request_id} "
+                                    f"queue={self.request_queue_properties.name} "
+                                    f"correlation_id={correlation_id} "
+                                    f"error='{UNDECODABLE}'")
+                    raise ResponseException(message=UNDECODABLE)
+                else:
+                    logging.info(f"event=requestCompleted "
+                                 f"client={self.name} "
+                                 f"id={request_id} "
+                                 f"queue={self.request_queue_properties.name} "
+                                 f"correlation_id={correlation_id}")
+                    return response
+            else:
+                if wait_for_response_time is None:
+                    wait_for_response_time = time.time()
+                    await asyncio.sleep(0)
+                elif time.time() - wait_for_response_time >= self.max_latency:
+                    logging.warning(f"event=requestError "
+                                    f"client={self.name} "
+                                    f"id={request_id} "
+                                    f"queue={self.request_queue_properties.name} "
+                                    f"correlation_id={correlation_id} "
+                                    f"error='{TIMED_OUT}'")
+                    self._terminate_correlation_id(correlation_id)
+                    raise ResponseException(message=TIMED_OUT)
```

Comment:
This method is pretty long! What about breaking it into a publish part, a handle-response part, and a wait-for-response part?